AdyenCheckout/AdyenCheckoutComponent.swift

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright (c) 2025 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@_spi(AdyenInternal) import Adyen
+#if canImport(AdyenSession)
+    @_spi(AdyenInternal) import AdyenSession
+#endif
+#if canImport(AdyenDropIn)
+    @_spi(AdyenInternal) import AdyenDropIn
+#endif
+#if canImport(AdyenComponents)
+    @_spi(AdyenInternal) import AdyenComponents
+#endif
+#if canImport(AdyenActions)
+    @_spi(AdyenInternal) import AdyenActions
+#endif
+
+// TODO: add description
+public final class AdyenCheckoutComponent {
+    
+    private var paymentComponent: PaymentComponent?
+    
+    private var actionComponent: ActionComponent?
+    
+    private var actionHandlingComponent: ActionHandlingComponent?
+    
+    private var configuration: CheckoutConfiguration
+    
+    package init(
+        paymentMethod: PaymentMethod,
+        configuration: CheckoutConfiguration,
+        session: AdyenSession? = nil
```

Comment:
Perhaps we can create a more generic "AnySession" in `AdyenCore` (that would represent an advanced mode); this way, the `AdyenCheckout` module would not need to directly depend on the `AdyenSession` module.

Reply:
We will change the internal code of session for sure. It won't have some of the stuff now that session itself won't be accessible